lib/teams/server/message-actions/chat-message-actions.js

```diff
@@ -0,0 +1,235 @@
+            let playerJoining = sc.get(props, 'playerJoining', false);
+            if (!playerJoining) {
+                return false;
+            }
+            let clan = sc.get(props, 'clan', false);
+            if (!clan) {
+                return false;
+            }
+            let playerJoiningName = playerJoining.playerName;
+            let message = this.createMessage(
+                TeamsConst.CHAT.MESSAGE.INVITE_ACCEPTED,
+                playerJoiningName
+            );
+            this.sendMessage(message, 'Clan', clan.ownerClient);
+        });
+    }
+
+    inviteClanRejectedEventListener()
+    {
+        this.events.on('reldens.clanJoinInviteRejected', (props) => {
+            let clientSendingInvite = sc.get(props, 'clientSendingInvite', false);
+            let playerRejectingName = sc.get(props, 'playerRejectingName', false);
+            if (!clientSendingInvite ||
+                !playerRejectingName) {
+                return false;
+            }
+            let message = this.createMessage(
+                TeamsConst.CHAT.MESSAGE.INVITE_REJECTED,
+                playerRejectingName
+            );
+            this.sendMessage(message, 'Clan', clientSendingInvite);
+        });
+    }
+
+    clanMemberLeavingEventListener()
+    {
+        this.events.on('reldens.clanLeaveBeforeSendUpdate', (props) => {
+            let playerLeavingId = sc.get(props, 'playerId', false);
+            if (!playerLeavingId) {
+                Logger.error('playerLeavingId undefined on ChatMessageAction');
+                return false;
+            }
+            let currentClan = sc.get(props, 'currentClan', false);
+            if (!currentClan) {
+                Logger.error('currentClan undefined on ChatMessageAction');
+                return false;
+            }
+            let disbandClan = sc.get(props, 'disbandClan', false);
+
+            if (disbandClan && playerLeavingId !== currentClan.owner.id) {
+                this.clanDisbanding(currentClan.owner.playerName, currentClan.clients[playerLeavingId]);
+                return true;
+            }
+            let message = this.createMessage(TeamsConst.CHAT.MESSAGE.LEAVE,
+                currentClan.players[playerLeavingId].playerName,
+                'clan');
+            this.sendMessage(message, 'Clan', currentClan.ownerClient);
+        });
+    }
+
+    clanDisbanding(clanOwnerName, clientLeaving)
+    {
+        let message = this.createMessage(TeamsConst.CHAT.MESSAGE.DISBANDED,
+            clanOwnerName,
+            'clan');
+        this.sendMessage(message, 'Clan', clientLeaving);
+    }
+
+    createMessage(baseChatMessage, playerName, groupName = false)
+    {
+        let message = baseChatMessage.replace('%playerName', playerName);
+        if (groupName) {
+            message = message.replace('%groupName', groupName);
+        }
+        return message;
+    }
+
+    sendMessage(message, chatFrom, client)
```

Comment:
check and IF POSSIBLE try to reuse the sendPrivateMessage or sendGlobalMessage from the RoomChat class, but I think even I made the same mistake of duplicating the funcionallity from it on the chat/server/message/[classes]